GTMAppAuthSwift/Sources/GTMKeychain.swift

```diff
@@ -15,9 +15,345 @@
+      accessibility: kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+    )
+  }
+
+  /// Retrieves the password `Data` for the given `String` identifier.
+  /// - parameter name: A `String` identifier for the Keychain item.
+  /// - returns: The password `Data` if found.
+  /// - throws: An instance of `KeychainWrapper.Error`.
+  func passwordData(forName name: String) throws -> Data {
+    try passwordDataFromKeychain(keychainItemName: name)
+  }
+
+  /// Retrieves the password `Data` for the given `String` identifier.
+  /// - parameter name: A `String` identifier for the Keychain item.
+  /// - parameter usingDataProtectionKeychain: A `Bool` indicating whether to
+  /// use the data protection keychain on macOS 10.15.
+  /// - returns: The password `Data` if found.
+  /// - throws: An instance of `KeychainWrapper.Error`.
+  @available(macOS 10.15, *)
+  func passwordData(
+    forName name: String,
+    useDataProtectionKeychain: Bool
+  ) throws -> Data {
+    try passwordDataFromKeychain(
+      keychainItemName: name,
+      useDataProtectionKeychain: useDataProtectionKeychain
+    )
+  }
+
+  private func passwordDataFromKeychain(
+    keychainItemName: String,
+    useDataProtectionKeychain: Bool = false
+  ) throws -> Data {
+    keychainHelper.useDataProtectionKeychain = useDataProtectionKeychain
+    return try keychainHelper.passwordData(service: keychainItemName)
+  }
+
+  /// Removes stored password string, such as when the user signs out.
+  /// - parameter name: The Keychain name for the item.
+  /// - throws: An instance of `KeychainWrapper.Error`.
+  func removePasswordFromKeychain(forName name: String) throws {
+    try removePasswordFromKeychain(keychainItemName: name)
+  }
+
+  /// Removes stored password string, such as when the user signs out. Note that if you choose to
+  /// start using the data protection keychain on macOS, any items previously created will not be
+  /// accessible without migration.
+  /// - parameter name: The Keychain name for the item.
+  /// - parameter usingDataProtectionKeychain: A Boolean value that indicates
+  /// whether to use the data protection keychain on macOS 10.15+.
+  /// - throws: An instance of `KeychainWrapper.Error`.
+  @available(macOS 10.15, *)
+  func removePasswordFromKeychain(
+    forName name: String,
+    usingDataProtectionKeychain: Bool
+  ) throws {
+    try removePasswordFromKeychain(
+      keychainItemName: name,
+      usingDataProtectionKeychain: usingDataProtectionKeychain
+    )
+  }
+
+  private func removePasswordFromKeychain(
+    keychainItemName: String,
+    usingDataProtectionKeychain: Bool = false
+  ) throws {
+    keychainHelper.useDataProtectionKeychain = usingDataProtectionKeychain
+    try keychainHelper.removePassword(service: keychainItemName)
+  }
+}
+
+// MARK: - Keychain helper
+
+/// A protocol defining the helper API for interacting with the Keychain.
+/// - note: This protocol's purpose is to help `GTMKeychain` be testable.
```

Comment:
Not sure what the semantics of the `Note` tag are, could it be integrated into the summary here?

Reply:
I'm just dropping it.